distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso

```diff
@@ -480,6 +482,53 @@ decimal_div = Base_Generator.lift_binary_op "DECIMAL_DIV" x-> y->
 decimal_mod = Base_Generator.lift_binary_op "DECIMAL_MOD" x-> y->
     x ++ " - FLOOR(CAST(" ++ x ++ " AS decimal) / CAST(" ++ y ++ " AS decimal)) * " ++ y
 
+replace_text_cs_all = Base_Generator.lift_ternary_op "REPLACE_TEXT_CS_ALL" t-> p-> r->
+    Builder.code "REPLACE(" ++ t ++ ", " ++ p ++ ", " ++ r ++ ")"
+
+replace_text_cs_only_first = Base_Generator.lift_ternary_op "REPLACE_TEXT_CS_ONLY_FIRST" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 1)"
+
+replace_text_ci_all = Base_Generator.lift_ternary_op "REPLACE_TEXT_CI_ALL" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 0, 'i')"
+
+replace_text_ci_only_first = Base_Generator.lift_ternary_op "REPLACE_TEXT_CI_ONLY_FIRST" t-> p-> r->
+    escaped_pattern = escape_regex_param p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ escaped_pattern ++ ", " ++ r ++ ", 1, 1, 'i')"
+
+replace_regex_cs_all = Base_Generator.lift_ternary_op "REPLACE_REGEX_CS_ALL" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 0)"
+
+replace_regex_cs_only_first = Base_Generator.lift_ternary_op "REPLACE_REGEX_CS_ONLY_FIRST" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 1)"
+
+replace_regex_ci_all = Base_Generator.lift_ternary_op "REPLACE_REGEX_CI_ALL" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 0, 'i')"
+
+replace_regex_ci_only_first = Base_Generator.lift_ternary_op "REPLACE_REGEX_CI_ONLY_FIRST" t-> p-> r->
+    pattern_string = transform_literal_param (_.pattern_string) p
+    Builder.code "REGEXP_REPLACE(" ++ t ++ ", " ++ pattern_string ++ ", " ++ r ++ ", 1, 1, 'i')"
+
+replace_column_cs_all = Base_Generator.lift_ternary_op "REPLACE_COLUMN_CS_ALL" t-> p-> r->
+    Builder.code "REPLACE(" ++ t ++ ", " ++ p ++ ", " ++ r ++ ")"
```

Comment:
This is a lot of similar operations.

I'd consider using the metadata mechanism from introduced in #7221 for this. For future we need a better way to pass metadata around, but for now we can use that mechanism and I think that would simplify this quite.

One downside is - its then harder to say what combinations of options are supported. Maybe we can just have a special Dialect method `are_replace_options_supported ...`?

Reply:
Using metadata now, so this is gone.